core/src/main/scala/org/http4s/parser/Rfc3986Rules.scala

```diff
@@ -0,0 +1,37 @@
+package org.http4s
+package parser
+
+import org.http4s.internal.parboiled2._
+
+trait Rfc3986Rules extends StringBuilding { self: Parser =>
```

Comment:
This is slowly replacing `Rfc3986Parser`.  When this series of PRs is done, that class will probably be gone.